clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -288,6 +288,59 @@ public void close() {
         };
     }
 
+    /**
+     * It call simplifiedreadFrom() at class DefaultRecord.
+     * @return
+     */
+    private Iterator<Record> simplifiedIterator() {
+        if (count() == 0)
+            return Collections.emptyIterator();
+        try (CloseableIterator<Record> iterator = simplifiedCompressedIterator(BufferSupplier.NO_CACHING)) {
+            List<Record> records = new ArrayList<>(count());
+            while (iterator.hasNext())
+                records.add(iterator.next());
+            return records.iterator();
+        }
+    }
+
+    private CloseableIterator<Record> simplifiedCompressedIterator(BufferSupplier bufferSupplier) {
```

Comment:
We can have some code reuse by augmenting `compressedIterator` with an additional parameter like `skipKeyValue`, and then depend on that parameter have a conditional in `readNext` below. Then we can save not having this function any more.

Reply:
Also we can save the above `simplifiedIterator` similarly as well.